Tuples cannot be used in netcoreapp1.0

The following test if compiled for netcoreapp1.0

```C#
      [Fact]
        public async Task TestCSharpValueTupleUsage()
        {
            var testCode = @"
class TestClass {
  (int x, int y) TestMethod() { return (0, 1); }
}
";

            await new CSharpTest { TestCode = testCode }.RunAsync();
        }
```

Will fail with the following errors:

```ini
Microsoft.CodeAnalysis.Testing.CompilerErrorTests.TestCSharpValueTupleUsage [FAIL]
      System.InvalidOperationException : Mismatch between number of diagnostics returned, expected "0" actual "3"
      
      Diagnostics:
      // Test0.cs(3,3): error CS8137: Cannot define a class or member that utilizes tuples because the compiler required type 'System.Runtime.CompilerServices.TupleElementNamesAttribute' cannot be found. Are you missing a reference?
      // Test0.cs(3,3): error CS8179: Predefined type 'System.ValueTuple`2' is not defined or imported, or is declared in multiple referenced assemblies
      // Test0.cs(3,40): error CS8179: Predefined type 'System.ValueTuple`2' is not defined or imported, or is declared in multiple referenced assemblies
      
      
      Stack Trace:
        C:\source\roslyn-sdk\src\Tools\Microsoft.CodeAnalysis.Testing\Microsoft.CodeAnalysis.Analyzer.Testing\DefaultVerifier.cs(41,0): at Microsoft.CodeAnalysis.Testing.DefaultVerifier.Equal[T](T expected, T actual, String message)
        C:\source\roslyn-sdk\src\Tools\Microsoft.CodeAnalysis.Testing\Microsoft.CodeAnalysis.Analyzer.Testing\AnalyzerTest`1.cs(191,0): at Microsoft.CodeAnalysis.Testing.AnalyzerTest`1.VerifyDiagnosticResults(IEnumerable`1 actualResults, ImmutableArray`1 analyzers, DiagnosticResult[] expectedResults)
        C:\source\roslyn-sdk\src\Tools\Microsoft.CodeAnalysis.Testing\Microsoft.CodeAnalysis.Analyzer.Testing\AnalyzerTest`1.cs(150,0): at Microsoft.CodeAnalysis.Testing.AnalyzerTest`1.<VerifyDiagnosticsAsync>d__46.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
        C:\source\roslyn-sdk\src\Tools\Microsoft.CodeAnalysis.Testing\Microsoft.CodeAnalysis.Analyzer.Testing\AnalyzerTest`1.cs(133,0): at Microsoft.CodeAnalysis.Testing.AnalyzerTest`1.<RunAsync>d__45.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
        C:\source\roslyn-sdk\tests\Microsoft.CodeAnalysis.Testing\Microsoft.CodeAnalysis.Analyzer.Testing.UnitTests\CompilerErrorTests.cs(225,0): at Microsoft.CodeAnalysis.Testing.CompilerErrorTests.<TestCSharpValueTupleUsage>d__10.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
  Finished:    Microsoft.CodeAnalysis.Analyzer.Testing.UnitTests
```

--- a/src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs
+++ b/src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs
@@ -551,6 +551,11 @@ namespace Microsoft.CodeAnalysis.Testing
                 solution = solution.AddMetadataReference(projectId, MetadataReferences.MicrosoftVisualBasicReference);
             }
 
+            if (MetadataReferences.MscorlibFacadeReference != null)
+            {
+                solution = solution.AddMetadataReference(projectId, MetadataReferences.MscorlibFacadeReference);
+            }
+
             if (MetadataReferences.SystemRuntimeReference != null)
             {
                 solution = solution.AddMetadataReference(projectId, MetadataReferences.SystemRuntimeReference);

--- a/src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/MetadataReferences.cs
+++ b/src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/MetadataReferences.cs
@@ -1,12 +1,13 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
 using System.Linq;
 using System.Reflection;
 using Microsoft.CodeAnalysis;
 
-#if !NETSTANDARD1_5
-using System;
+#if NETSTANDARD1_5
+using System.IO;
 #endif
 
 namespace Microsoft.CodeAnalysis.Testing
@@ -23,8 +24,9 @@ namespace Microsoft.CodeAnalysis.Testing
         public static readonly MetadataReference SystemCollectionsImmutableReference = MetadataReference.CreateFromFile(typeof(ImmutableArray).GetTypeInfo().Assembly.Location);
         public static readonly MetadataReference MicrosoftVisualBasicReference = MetadataReference.CreateFromFile(typeof(Microsoft.VisualBasic.Strings).GetTypeInfo().Assembly.Location);
 
-        public static readonly MetadataReference SystemRuntimeReference;
-        public static readonly MetadataReference SystemValueTupleReference;
+        internal static readonly MetadataReference MscorlibFacadeReference;
+        internal static readonly MetadataReference SystemRuntimeReference;
+        internal static readonly MetadataReference SystemValueTupleReference;
 
         static MetadataReferences()
         {
@@ -32,14 +34,25 @@ namespace Microsoft.CodeAnalysis.Testing
             if (typeof(string).GetTypeInfo().Assembly.ExportedTypes.Any(x => x.Name == "System.ValueTuple"))
             {
                 // mscorlib contains ValueTuple, so no need to add a separate reference
+                MscorlibFacadeReference = null;
                 SystemRuntimeReference = null;
                 SystemValueTupleReference = null;
             }
+            else
+            {
+                MscorlibFacadeReference = MetadataReference.CreateFromFile(Path.Combine(Path.GetDirectoryName(typeof(Lazy<,>).GetTypeInfo().Assembly.Location), "mscorlib.dll"));
+                SystemRuntimeReference = MetadataReference.CreateFromFile(typeof(Lazy<,>).GetTypeInfo().Assembly.Location);
+                SystemValueTupleReference = MetadataReference.CreateFromFile(typeof(ValueTuple<,>).GetTypeInfo().Assembly.Location);
+            }
 #elif NETSTANDARD2_0
             // mscorlib contains ValueTuple, so no need to add a separate reference
+            MscorlibFacadeReference = null;
             SystemRuntimeReference = null;
             SystemValueTupleReference = null;
 #elif NET452
+            // System.Object is already in mscorlib
+            MscorlibFacadeReference = null;
+
             var systemRuntime = AppDomain.CurrentDomain.GetAssemblies().SingleOrDefault(x => x.GetName().Name == "System.Runtime");
             if (systemRuntime != null)
             {

--- a/src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/PublicAPI.Unshipped.txt
+++ b/src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/PublicAPI.Unshipped.txt
@@ -132,8 +132,6 @@ static readonly Microsoft.CodeAnalysis.Testing.MetadataReferences.MicrosoftVisua
 static readonly Microsoft.CodeAnalysis.Testing.MetadataReferences.SystemCollectionsImmutableReference -> Microsoft.CodeAnalysis.MetadataReference
 static readonly Microsoft.CodeAnalysis.Testing.MetadataReferences.SystemCoreReference -> Microsoft.CodeAnalysis.MetadataReference
 static readonly Microsoft.CodeAnalysis.Testing.MetadataReferences.SystemReference -> Microsoft.CodeAnalysis.MetadataReference
-static readonly Microsoft.CodeAnalysis.Testing.MetadataReferences.SystemRuntimeReference -> Microsoft.CodeAnalysis.MetadataReference
-static readonly Microsoft.CodeAnalysis.Testing.MetadataReferences.SystemValueTupleReference -> Microsoft.CodeAnalysis.MetadataReference
 virtual Microsoft.CodeAnalysis.Testing.AnalyzerTest<TVerifier>.ApplyCompilationOptions(Microsoft.CodeAnalysis.Project project) -> Microsoft.CodeAnalysis.Project
 virtual Microsoft.CodeAnalysis.Testing.AnalyzerTest<TVerifier>.CreateProjectImpl((string filename, Microsoft.CodeAnalysis.Text.SourceText content)[] sources, (string filename, Microsoft.CodeAnalysis.Text.SourceText content)[] additionalFiles, Microsoft.CodeAnalysis.MetadataReference[] additionalMetadataReferences, string language) -> Microsoft.CodeAnalysis.Project
 virtual Microsoft.CodeAnalysis.Testing.AnalyzerTest<TVerifier>.CreateSolution(Microsoft.CodeAnalysis.ProjectId projectId, string language) -> Microsoft.CodeAnalysis.Solution

--- a/tests/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing.UnitTests/CompilerErrorTests.cs
+++ b/tests/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing.UnitTests/CompilerErrorTests.cs
@@ -213,8 +213,6 @@ End Class
             await new VisualBasicTest { TestCode = testCode }.RunAsync();
         }
 
-#if NETCOREAPP2_0 || NET46
-// Disable this test for  netcorapp1.0 as tuples cannot be resolved
         [Fact]
         public async Task TestCSharpValueTupleUsage()
         {
@@ -226,7 +224,6 @@ class TestClass {
 
             await new CSharpTest { TestCode = testCode }.RunAsync();
         }
-#endif
 
         private class CSharpTest : AnalyzerTest<DefaultVerifier>
         {

